picard/file.py

```diff
@@ -98,12 +98,26 @@ def new_metadata(self):
 
     def load(self, callback):
         thread.run_task(
-            partial(self._load, self.filename),
+            partial(self._load_check, self.filename),
             partial(self._loading_finished, callback),
             priority=1)
 
+    def _load_check(self, filename):
+        # Check that file has not been removed since thread was queued
+        # Don't load if we are stopping.
+        if self.state != File.PENDING or self.tagger.stopping:
```

Comment:
At the expense of a minor code duplication, for the sake of clarity, please rewrite it as:

```python
if self.state != File.PENDING:
    log.debug("File not loaded because it was removed: %r", self.filename)
    return None
if self.tagger.stopping:
    log.debug("File not loaded because %s is stopping: %r", PICARD_APP_NAME, self.filename)
    return None
return self._load(filename)
```

Also don't use hardcoded Picard in strings (well, i guess there are already too many occurences), we have a constant for it. It will ease a fork under another name (open source!!). Or just use "application" instead of "Picard).

Also, we may want to translate those strings at some point, and building strings as you did here isn't very i18n friendly.